tuf/client/client.go

```diff
@@ -46,493 +42,197 @@ func (c *Client) Update() error {
 	err := c.update()
 	if err != nil {
 		logrus.Debug("Error occurred. Root will be downloaded and another update attempted")
-		if err := c.downloadRoot(); err != nil {
+		logrus.Debug("Resetting the TUF builder...")
+
+		repoSoFar := c.builder.GetRepo()
+		c.builder = c.builder.BootstrapNewBuilder()
```

Comment:
Trying to figure out why the existing builder couldn't be used.

I see now that we essentially want to restart the build process at this point, because we have more information. (That appears to be the point of `BootstrapNewBuilder`, as far as I can tell)